sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client.py

```diff
@@ -1115,31 +1155,21 @@ def get_ranges(  # type: ignore
         :paramtype lease: ~azure.storage.fileshare.ShareLeaseClient or str
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
-        :returns: A list of valid ranges.
-        :rtype: List[dict[str, int]]
+        :returns:
+            A tuple of two lists of file ranges as dictionaries with 'start' and 'end' keys.
+            The first element are filled file ranges, the 2nd element is cleared file ranges.
+        :rtype: tuple(list(dict(str, str), list(dict(str, str))
         """
-        timeout = kwargs.pop('timeout', None)
-        if self.require_encryption or (self.key_encryption_key is not None):
-            raise ValueError("Unsupported method for encryption.")
-        access_conditions = get_access_conditions(kwargs.pop('lease', None))
-
-        content_range = None
-        if offset is not None:
-            if length is not None:
-                end_range = offset + length - 1  # Reformat to an inclusive range index
-                content_range = 'bytes={0}-{1}'.format(offset, end_range)
-            else:
-                content_range = 'bytes={0}-'.format(offset)
+        options = self._get_ranges_options(
+            offset=offset,
+            length=length,
+            previous_sharesnapshot=previous_sharesnapshot,
+            **kwargs)
         try:
-            ranges = self._client.file.get_range_list(
-                range=content_range,
-                sharesnapshot=self.snapshot,
-                lease_access_conditions=access_conditions,
-                timeout=timeout,
-                **kwargs)
+            ranges = self._client.file.get_range_list(**options)
         except StorageErrorException as error:
             process_storage_error(error)
-        return [{'start': b.start, 'end': b.end} for b in ranges]
+        return get_file_ranges_result(ranges)
```

Comment:
this will be a breaking change for existing customers who are using get_ranges().
Currently the customers are receiving a List, after this change they will receive a Tuple.

Reply:
I have created a check to see if a snapshot is given to compare against. If there isn't then the API behaves as it used to, i.e, returns a list. If a snapshot is given then it will return the tuple with file_ranges and clear_ranges